services/advisoryValidator.js

```diff
@@ -0,0 +1,184 @@
+            const details = showDetails ? {"ruleName": itemWithUnmatchedCrossRefs.advisoryTestName} : {};
+            return {
+                "id": itemWithUnmatchedCrossRefs.fullAdvisoryDefinition.id,
+                "category": itemWithUnmatchedCrossRefs.fullAdvisoryDefinition.category,
+                "severity": itemWithUnmatchedCrossRefs.fullAdvisoryDefinition.severity,
+                "message":  itemWithUnmatchedCrossRefs.fullAdvisoryDefinition.message,
+                "identifier": getIatiIdentifierOfContainingActivity(itemWithUnmatchedCrossRefs.matchingElement) || 'noIdentifier',
+                "title": getIatiActivityTitleOfContainingActivity(itemWithUnmatchedCrossRefs.matchingElement) || 'noTitle',
+                "context": getFailureContextFromDOM(itemWithUnmatchedCrossRefs, adjustLineCountForMissingXmlTag),
+                details
+            };
+        });
+
+    return advisoryFailures;
+}
+
+function getElementsMatchingAdvisoryXPath(advisoryDefinitions, iatiDOM) {
+    
+    const elementsMatchingAdvisoryXPath = Object.values(advisoryDefinitions).map((advisoryDefinition) => {
+        const matchingElements = select(advisoryDefinition.advisoryXPath, iatiDOM);
+        return matchingElements.filter(match => {
+                                    let hasTargetAttribute = false;
+                                    for (let i = 0; i < match.attributes.length; i += 1) {
+                                        hasTargetAttribute ||= match.attributes[i].name === advisoryDefinition.attributeToTest;
+                                    } 
+                                    return hasTargetAttribute;
+                               })
+                               .map(matchingElement => ({
+                                    "advisoryTestName": advisoryDefinition.advisoryTestName, 
+                                    "advisoryXPath": advisoryDefinition.advisoryXPath,
+                                    "fullAdvisoryDefinition": advisoryDefinition,
+                                    "matchingElement": matchingElement
+                               }));
+    }).flat();
+
+    return elementsMatchingAdvisoryXPath;
+}
+
+
+// eslint-disable-next-line no-unused-vars
+export default async function validateAdvisories(iatiVersion, iatiXml, adjustLineCountForMissingXmlTag, showDetails = false) {
+    const advisories = [];
+    const advisoryDefinitions = getAdvisoryDefinitions(iatiVersion);
+    const advisoryHandlers = {'iati_identifier_exists': checkIatiIdentifierCrossReferences};
+
+    const iatiDOM = new DOMParser().parseFromString(iatiXml);
+
+    const elementsMatchingAdvisoryXPath = getElementsMatchingAdvisoryXPath(advisoryDefinitions, iatiDOM);
+
+    const elementsMatchingAdvisoryGrouped = groupListIntoMap(elementsMatchingAdvisoryXPath, 
+                                                             advisory => advisory.advisoryTestName);
+
+    // eslint-disable-next-line no-restricted-syntax
```

Comment:
Is there a specific reason for disabling this and the rule?

Reply:
It's because `forEach` doesn't support `async`/`await`, so I used a `for` loop instead, but `eslint` in this configuration is against `for` loops.